### ISSUE TITLE
Multi Datacenter Features?

### ISSUE THREAD

**Issue by brianshumate** [ansible-collections/ansible-consul#67]
This is just a discussion/thinking issue around what kinds of support we might like to add in for multiple datacenters given the initial bits to differentiate between LAN and WAN servers which were recently added by @groggemans.

I'd like to get some ideas about any features which would be helpful to add, like:

- WAN joining new or existing clusters (a dedicated playbook for this?)
- [Basic Federation with the WAN Gossip Pool](https://www.consul.io/docs/guides/datacenters.html)
- [Advanced Federation with Network Areas](https://www.consul.io/docs/guides/areas.html) for Consul Enterprise users

If there are ideas for what would be handy and how we might implement it, let's coordinate here before hitting the YAML and so on. 😄

**Comment by groggemans** [comment]
Something I was thinking about while implementing the basic support is different WAN/LAN ip's. A lot of production setups probably need this.

**Comment by brianshumate** [comment]
Indeed, that's a good point for sure. I would like to continue to make the role more usable for various production setups and that would definitely help.

**Comment by bbaassssiiee** [comment]
Adding the Mesh gateway would be very cool.
